File: BinarySearch/jichon/2417.js
const fs = require("fs");
const input = fs.readFileSync("./input.txt").toString().trim().split("\n");

function binary(n) {
  let start = 0;
  let end = n;
  while (start <= end) {
    mid = Math.floor((start + end) / 2);
    if (mid * mid < n) start = mid + 1;
    else end = mid - 1;
  }
  return start;
}
const solution = () => {
  let n = +input[0];
  console.log(binary(n));
};

solution();

File: BinarySearch/박연주/2417.py
def binarySearch(start, end):

    if(start > end):
        return start
    
    mid = (start + end) // 2
    target = mid ** 2
    
    if target == n:
        return mid
    elif target < n:
        return binarySearch(mid + 1, end)
    else:
        return binarySearch(start, mid - 1)
    
n = int(input())
start = 0
end = n

print(binarySearch(start, end))